### PR TITLE
Cast similarities to probabilities using tf.softmax(similarities)

### PR DESCRIPTION
**Pull request by hyperji**

### CODE REVIEW

**Comment by AntreasAntoniou** [comment]
This is not necessary tf adds the softmax in the loss function. If you add that you'll notice that you have much higher losses because you take the softmax of the softmax. An earlier version was mistakenly like that.

**Comment by hyperji** [comment]
Thanks for reply, I now understand. :-)

**Comment by hyperji** [comment]
@AntreasAntoniou Thanks for reply, I now understand. :-)
